README.md

```diff
@@ -1,93 +1,134 @@
 # tips
+> Tips, hacks and guidelines for optimizing JavaScript modules
 
-In general, your goal is to convert as many [statements](https://github.com/estree/estree/blob/master/es5.md#statements) as you can into [expressions](https://github.com/estree/estree/blob/master/es5.md#expressions).
+Above all else, `baggo` modules should prioritize being fast, compact and simple. Therefore, single variable names and minimal whitespace are  acceptable when writing `baggo`-style code.
 
-## Reuse Old Variables
+Regardless, whitespace has been included in some of the following examples for clarity.
 
-Create a few "`var` chains" and tag on extra declarations:
+## Optimizing loops
+A loop is a block of code that is repeated until a specified condition is fulfilled. The three main methods of looping in JavaScript (and other similar languages) are `for` loops, `while` loops and recursion.
 
-```js
-var x=123,y=345,z,a,b
+`while` loops focus primarily on readability, and recursion is slow in JavaScript. Therefore, the _only_ kind of loop you should be concerned with when writing `baggo` modules is the `for` loop.
+
+When writing a loop in which the current iteration index `i` is used, most programmers will write a `for` loop like this:
+
+```javascript
+// for (initial; condition; iteration);
+for (var i = 0; i < array.length; i++) {
+  var item = array[i]
+  // Do stuff with item...
+}
 ```
 
-This way, you don't have to use `var` later on.
+Unless `array.length` is constantly changing, it is preferable to cache it in a variable to avoid unnecessary property lookups.
 
-**Note:** Be careful with scoping.  Use multiple variables to get past conflicts.
+```javascript
+// In this example, `array.length` is only accessed once
+for (var i = 0, max = array.length; i < max; i++) {
+  // ...
+}```
 
-## Smallest Loop
+An even faster and more compact approach would be to loop backwards through the array:
 
-```js
-for(init;condition;body);
+```javascript
+for (var i = array.length; i--;) {
+  // ...
+}
 ```
 
-For infinite loops just leave them empty:
+Other than being fast, this method has an additional perk - if you need to remove elements as you iterate, looping backwards may eliminate the need to implement additional low-level logic to accommodate item removal.
 
-```js
-for(;;);
-```
+It's also possible to move all the code between the braces into the final `iteration` expression of the `for` loop if need be. However, note that common keywords and tokens such as `if` and `break` cannot be used in the middle of expressions. For complex logic, use [ternaries and logical operators](#condensing-conditionals) instead.
```

Comment:
Same nit-pick about the comment below.

These are called "conditional expressions" and "logical expressions".  I think it would be important to create distinguished ideas from statements and expressions.  In my head its easier to compare them that way and find alternatives.